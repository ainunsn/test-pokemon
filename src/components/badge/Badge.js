/** @jsxImportSource @emotion/react */

import React from "react";
import { capitalize } from "utils";

function Badge({ value }) {
  return (
    <p
      data-testid="custom-badge"
      css={{
        textAlign: "center",
        backgroundColor: "#0e61a7",
        borderRadius: "50px",
        color: "white",
        padding: "0.2em 1.2em",
      }}
      className="bagde"
    >
      {value && capitalize(value)}
    </p>
  );
}

export default Badge;
