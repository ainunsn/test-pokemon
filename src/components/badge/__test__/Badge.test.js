import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Badge from "../Badge";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");

  ReactDOM.render(<Badge />, div);
});

it("renders badge correctly", () => {
  const { getByTestId } = render(<Badge value={"value"} />);
  expect(getByTestId("custom-badge")).toHaveTextContent("Value");
});

// snapshot test
it("match snapshot", () => {
  const tree = renderer.create(<Badge value={"value"} />);

  expect(tree).toMatchSnapshot();
});
