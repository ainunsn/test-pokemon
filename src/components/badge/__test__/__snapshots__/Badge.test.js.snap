// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<p
  className="bagde css-10ut616-Badge"
  data-testid="custom-badge"
>
  Value
</p>
`;
